workers/unity/Packages/io.improbable.gdk.buildsystem/Configuration/BuildContext.cs

```diff
@@ -14,60 +12,30 @@ public struct BuildContext
         public ScriptingImplementation ScriptingImplementation;
         public iOSSdkVersion? IOSSdkVersion;
 
-        public static List<BuildContext> GetBuildContexts(IEnumerable<string> wantedWorkerTypes,
-            BuildEnvironment buildEnvironment, ScriptingImplementation? scriptImplementation = null,
-            ICollection<BuildTarget> buildTargetFilter = null, iOSSdkVersion? iosSdkVersion = null)
+        public static List<BuildContext> GetBuildContexts(BuildContextFilter contextFilter)
         {
             var spatialOsBuildConfiguration = BuildConfig.GetInstance();
```

Comment:
In order to test properly, this should be injected as well right? Currently the tests will use whatever build configuration is defined in the project which: 

- doesn't allow us to construct edge cases to test
- means our tests are brittle and can break because we changed the default build configuration

Reply:
Correct - but I'm not sure how I'd inject our own BuildConfig if it's a singleton scriptable object

Reply:
The `SingletonScriptableObject` allows you to make new versions of the assets and doesn't count it as long as its not saved to disk (i.e. - in memory only).

https://github.com/spatialos/gdk-for-unity/blob/6b8e88650b7132c83120ce642e2fd39b7f63e846/workers/unity/Packages/io.improbable.gdk.core/Editor/SingletonScriptableObject.cs#L28

You should be able to use the `CreateInstance` method as normal if I'm looking at this correctly.

Reply:
ah nice, I'll look into it